shippers/shipper.py

```diff
@@ -30,3 +30,11 @@ def flush(self) -> None:
         """
 
         raise NotImplementedError
+
+    @abstractmethod
+    def discover_dataset(self, event: dict[str, Any]) -> None:
```

Comment:
`discover_dataset` is strictly relevant to `ElaticsearchShipper` implementation and should not be defined in the abstract class

Reply:
agreed, but if I remove the abstract method, when calling at `handlers.aws.handlers.py: 63 shipper.discover_dataset(event=lambda_event)`, linting will complain because `ShipperFactory.create_from_output` returns shipper of type CommonShipper which is an abstract class.

Should I refactor ShipperFactory to return an ElasticsearchShipper if the output_type is elasticsearch like in our case?

Reply:
you can overcome that introducing a generic like:
`CommonShipperType = TypeVar("CommonShipperType", bound=CommonShipper)`
in https://github.com/elastic/elastic-serverless-forwarder/blob/main/shippers/shipper.py

then you can do:
`shipper: ElasticsearchShipper = ShipperFactory.create_from_output(`

and `ShipperFactory.create_from_output()` should return the generic (as well as `ShipperFactory.create()`)